### PR TITLE
Updating contributing docs, mock_with Rspec for future compatibility, and puppet-vault_lookup dependency

### PR DESCRIPTION
**Pull request by ekingme**
Updating contributing docs, mock_with Rspec for future compatibility, and puppet-vault_lookup dependency

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,11 +17,10 @@ will likely make it into a release a little quicker.
 
 5. Push to your fork and submit a pull request.
 
-
 ## Dependencies
 
-The testing and development tools have a bunch of dependencies,
-all managed by [Bundler](http://bundler.io/) according to the
+This module is developed using PDK, and all dependencies are managed by
+[Bundler](http://bundler.io/) provided through PDK aoccording to the
```
Comment (rnelson0) [inline CONTRIBUTING.md:23]:
```suggestion
[Bundler](http://bundler.io/) provided through PDK according to the
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -18,11 +17,10 @@ will likely make it into a release a little quicker.
...
 [Puppet support matrix](http://docs.puppetlabs.com/guides/platforms.html#ruby-versions).
```
Comment (rnelson0) [inline CONTRIBUTING.md:24]: How about removing L24-36 and replacing with a link to https://puppet.com/docs/pdk/1.x/pdk_testing.html ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -34,16 +32,15 @@ you must set an environment variable such as:
...
-The test suite will run [Puppet Lint](http://puppet-lint.com/) and
-[Puppet Syntax](https://github.com/gds-operations/puppet-syntax) to
-check various syntax and style things. You can run these locally with:
+You can test style and syntax of your module using PDK.  This will run tests
+against your manifests, templates, metadata, and spec tests using rubocop,
+[Puppet Lint](http://puppet-lint.com/) and [Puppet Syntax](https://github.com/gds-operations/puppet-syntax).
 
-    bundle exec rake lint
-    bundle exec rake syntax
+    pdk validate
```
Comment (rnelson0) [inline CONTRIBUTING.md:43]:
```suggestion
    pdk validate

If any issues are found, the PDK can perform auto-corrections with the `-a` flag

    pdk validate -a
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -53,40 +50,8 @@ add tests if you're adding new functionality. If you've not used
 about how best to test your new feature. Running the test suite is done
 with:
 
-    bundle exec rake spec
+    pdk bundle exec rake spec
```
Comment (rnelson0) [inline CONTRIBUTING.md:53]:
```suggestion
    pdk test unit
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -53,40 +50,8 @@ add tests if you're adding new functionality. If you've not used
...
 
 Note also you can run the syntax, style and unit tests in one go with:
```
Comment (rnelson0) [inline CONTRIBUTING.md:55]: Delete L55-57

File: metadata.json [hunk 6]
```diff
@@ -8,7 +8,10 @@
   "project_page": "https://github.com/rnelson0/puppet-certs",
   "issues_url": "https://github.com/rnelson0/puppet-certs/issues",
   "dependencies": [
-
+    {
```
Comment (rnelson0) [inline metadata.json:11]: I think I'd prefer to have this a soft dependency and documented in the readme. Something like

```
To use the vault options, you must have a module that is API compatible with <link to forge for vault_lookup> installed. If you are not using vault, this dependency is optional.
```

**Review by rnelson0 — CHANGES_REQUESTED** [review]

**Review by codylane — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (ekingme) [reply CONTRIBUTING.md:59]: This has been incorporated

File: metadata.json (hunk 6 above)
Comment (ekingme) [reply metadata.json:11]: This has been incorporated

**Review by codylane — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -18,12 +17,11 @@ will likely make it into a release a little quicker.
...
 
-
 ## Dependencies
 
-The testing and development tools have a bunch of dependencies,
-all managed by [Bundler](http://bundler.io/) according to the
-[Puppet support matrix](http://docs.puppetlabs.com/guides/platforms.html#ruby-versions).
+This module is developed using PDK, and all dependencies are managed by
+[Bundler](http://bundler.io/) provided through PDK according to the [best
+pracrices](https://puppet.com/docs/pdk/1.x/pdk_testing.html).
```
Comment (rnelson0) [inline CONTRIBUTING.md:24]:
```suggestion
practices](https://puppet.com/docs/pdk/1.x/pdk_testing.html).
```

File: metadata.json (hunk 6 above)
Comment (rnelson0) [inline metadata.json:11]: Looks like this got left in, probably because I haven't figured out how to do multi-line suggestions.

**Review by rnelson0 — CHANGES_REQUESTED** [review]
Minor changes left.

**Review by rnelson0 — APPROVED** [review]

**Pull request merged**